src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -608,6 +608,32 @@ theorem shiftLeftZeroExtend_eq {x : BitVec w} :
     (shiftLeftZeroExtend x i).msb = x.msb := by
   simp [shiftLeftZeroExtend_eq, BitVec.msb]
 
+theorem BitVec.shiftLeft_shiftLeft (w : Nat) (x : BitVec w) (n m : Nat) :
+    (x <<< n) <<< m = x <<< (n + m) := by
+  ext i
+  simp only [getLsb_shiftLeft, Fin.is_lt, decide_True, Bool.true_and]
+  -- omega claims that the system cannot be proven, so we case bash.
+  -- The entire proof below should be redundant once omega is complete.
```

Comment:
`decide` statements are out of scope for `omega`!

Reply:
If you want to case bash and then hand off to `omega`, use:
```
theorem BitVec.shiftLeft_shiftLeft (w : Nat) (x : BitVec w) (n m : Nat) :
    (x <<< n) <<< m = x <<< (n + m) := by
  ext i
  simp only [getLsb_shiftLeft, Fin.is_lt, decide_True, Bool.true_and]
  rw [show i - (n + m) = (i - m - n) by omega]
  cases h₂ : decide (i < m) <;> cases h₃ : decide (i - m < w) <;> cases h₄ : decide (i - m < n) <;> cases h₅ : decide (i < n + m) <;>
    simp at * <;> omega
```

Reply:
Sweet, thanks for the tutorial! `rw [show _ by omega]` is a neat one, I'm going to remember this golf trick :)